### ISSUE TITLE
Instructions for Custom HTML tag handling do not work (or: Custom HTML tag handling does not work)

### ISSUE THREAD

**Issue by simonpercivall** [weblyzard/inscriptis#81]
In `Inscriptis.__init__`, `start_tag_handler_dict` and `_end_tag_handler_dict` are set, and at the end of init, `._parse_html_tree` is called, going through the tree and calling the handlers.

This means that setting handlers after instantiating `Inscriptis` does nothing, since the tree has already been handled. And since the handlers are set in `__init__`, it's also not possible to set them on a subclass, and subclassing `Inscriptis` and overriding `__init__` means having to call `self._parse_html_tree()` twice.

Thanks!

**Comment by AlbertWeichselbraun** [comment]
thank you @simonpercivall for spotting this. 

Can you please take a short look at https://github.com/weblyzard/inscriptis/tree/fix/bug-81-custom-html-handling and the example provided in https://github.com/weblyzard/inscriptis/blob/fix/bug-81-custom-html-handling/examples/custom-html-handling.py which should fix this issue?

**Comment by simonpercivall** [comment]
Hi! Thanks a lot for the quick response to this!

It certainly looks like that change is a good fix to the issue.

I only hope there's no problems with `inscriptis.canvas` not having any content before calling `inscriptis.get_text()`; it feels like a possible regression, depending on whether you see the `canvas` attribute as being an implementation detail or a public attribute.

**Comment by AlbertWeichselbraun** [comment]
you are right - we probably need to redesign the class to make this really robust.

**Comment by simonpercivall** [comment]
An alternative could be to give either a way to pass the handlers as argument to `__init__`, or a method to override that `__init__` calls after populating the defaults.

Thanks again!

**Comment by AlbertWeichselbraun** [comment]
I changed the design to allow defining custom html handler through `ParserConfig`. 

see https://github.com/weblyzard/inscriptis/tree/fix/bug-81-custom-html-handling2 and adjusted the corresponding [example code](https://github.com/weblyzard/inscriptis/blob/fix/bug-81-custom-html-handling2/examples/custom-html-handling.py).

please let me know, if you see anything problematic. otherwise i would draft a new release which incorporates these changes.

**Comment by simonpercivall** [comment]
I can look at it properly on Monday, but from what I see, it looks great! Thanks for spending time on this!

**Comment by AlbertWeichselbraun** [comment]
excellent - than i will wait for your feedback.
